### PR TITLE
Implement search celebrities

### PR DESCRIPTION
**Pull request by MowJay**

### CODE REVIEW

**Comment by farnoodma** [comment]
Thank you @MohammadJB for your PR. Would you please resolve the conflicts?

**Comment by MowJay** [comment]
Conflicts resolved and SearchBox component improved @farnoodma

File: src/components/SearchInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
+export type SearchInputProps = {
+  value: string
+  handleChange: (value: string) => void
+}
+
+const SearchInput = ({ value, handleChange }: SearchInputProps) => {
+  const onChange = (event: { target: { value: string } }) => {
```
Comment (blackwolf202) [inline src/components/SearchInput/index.tsx:9]:
```suggestion
  const onChange = (event) => {
```

File: src/components/SearchInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
```
Comment (blackwolf202) [inline src/components/SearchInput/index.tsx:1]: nit: better to have File name as 
`src/components/SearchInput/SearchInput.tsx`
and have index.ts as: 
` export {SearchInput} from './SearchInput'; `

File: src/components/Grid/index.tsx [hunk 3]
```diff
@@ -3,20 +3,27 @@ import React from 'react'
...
   const renderBoxes = () =>
-    celebrities.map(celebrity => (
-      <Card
-        celebrity={celebrity as Celebrity}
-        level={level}
-        visible={visible}
-      />
-    ))
+    celebrities.map(celebrity => {
+      if (!celebrity.name.toLowerCase().includes(searchText.toLowerCase()))
```
Comment (blackwolf202) [inline src/components/Grid/index.tsx]: IMO, a bit hard to follow what this bit of code is trying to do. 
Maybe pull the logic out into a variable: 
```
const hasSearchText = searchText !== '' && celebrity.name.toLowerCase().includes(searchText.toLowerCase())
```
and in your if
```
if(!hasSearchText) {
    return null;
} 
```
Note that `searchText !== '' && ` bit helps with readability and follows a mindset of being specific. This will help maintaining the code in future.
Essentially you do the `includes` check only when user has started searching

**Review by blackwolf202 — APPROVED** [review]
Minor suggestions, in general looks good

LGTM

File: src/components/SearchInput/index.tsx (hunk 1 above)
Comment (MowJay) [reply src/components/SearchInput/index.tsx:9]: Your suggestion will produce '@typescript-eslint/no-unsafe-argument' error

File: src/components/SearchInput/index.tsx (hunk 2 above)
Comment (MowJay) [reply src/components/SearchInput/index.tsx:1]: Thanks for your suggestion but I just wanted to keep the same file architecture

File: src/components/Grid/index.tsx (hunk 3 above)
Comment (MowJay) [reply src/components/Grid/index.tsx]: tnx, I used a variable to improve readability but I dont agree with `searchText !== '' &&` part because in that way if the searched text be empty, no item will be visible.
